configure.ac

```diff
@@ -0,0 +1,74 @@
+AC_INIT([chaosnet-bridge], 0.0)
+AC_CONFIG_AUX_DIR([build-aux])
+AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+
+AC_CONFIG_SRCDIR([cbridge.c])
+
+AC_ARG_ENABLE([etherp],
+    AS_HELP_STRING([--enable-etherp], [Enable Chaos-over-Ether code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-etherp])])],
+    [enable_etherp=yes])
+enable_etherp_val=0
+if test "x$enable_etherp" = "xyes"; then
+    enable_etherp_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_ETHERP], [$enable_etherp_val], [])
+AM_CONDITIONAL([ENABLE_ETHERP], [test x$enable_etherp = xyes])
+
+AC_ARG_ENABLE([tls],
+    AS_HELP_STRING([--enable-tls], [Enable Chaos-over-TLS code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-tls])])],
+    [enable_tls=yes])
+enable_tls_val=0
+if test "x$enable_tls" = "xyes"; then
+    enable_tls_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_TLS], [$enable_tls_val], [])
+AM_CONDITIONAL([ENABLE_TLS], [test x$enable_tls = xyes])
+
+AC_ARG_ENABLE([ip],
+    AS_HELP_STRING([--enable-ip], [Enable Chaos-over-IP code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-ip])])],
+    [enable_ip=yes])
+enable_ip_val=0
+if test "x$enable_ip" = "xyes"; then
+    enable_ip_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_IP], [$enable_ip_val], [])
+AM_CONDITIONAL([ENABLE_IP], [test x$enable_ip = xyes])
+
+# CHAOS_DNS // enable the DNS code (forwarder and internal host/addr parser)
+AC_ARG_ENABLE([dns],
+    AS_HELP_STRING([--enable-dns], [Enable DNS code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-dns])])],
+    [enable_dns=yes])
+enable_dns_val=0
+if test "x$enable_enable_dns" = "xyes"; then
+    enable_dns_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_DNS], [$enable_dns_val], [])
+AM_CONDITIONAL([ENABLE_DNS], [test x$enable_dns = xyes])
+
+#	CHAOS_DNS_SERVER
+#	CHAOS_ADDR_DOMAIN
+
+AC_PROG_CC
+AC_CHECK_HEADER([net/ethertypes.h],
+    [AC_DEFINE([HAVE_NET_ETHERTYPES_H], [1], 
+               [Define to 1 if you have <net/ethertypes.h>])])
+AC_CHECK_HEADER([net/ethernet.h], 
+     [AC_DEFINE([HAVE_NET_ETHERNET_H], [1], 
+                [Define to 1 if you have <net/ethernet.h>.])])
+AC_CHECK_HEADER([netpacket/packet.h],
+    [AC_DEFINE([HAVE_NETPACKET_PACKET_H], [1], 
+               [Define to 1 if you have <netpacket/packet.h>])])
+AC_HEADER_RESOLV
+AX_PTHREAD
```

Comment:
pthreads are always required (but then again I don't know what AX_PTHREAD does).

Reply:
AX_PTHREADS checks if pthreads is available on the system, and then sets CFLAGS, LDFLAGS, and stuff accordingly.  So on GNU/Linux systems, the right way to link to pthreads is not via -lpthreads only but also -pthreads, on other platforms it is adjusted accordingly.

Reply:
Here is a more detailed description of AX_PTHREAD: https://www.gnu.org/software/autoconf-archive/ax_pthread.html